woocommerce/woocommerce-gutenberg-products-block
# Update product category controls to use typescript

## What

`ProductControl` is using TypeScript already so I've removed the todo.
`ProductCategoryControl` was not using TypeScript, so I've converted it based on the above.

Fixes #6488
Fixes #6487

## Why

Refactoring/general improvements.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. In the post editor, insert some blocks which use category controls. `Featured Category`, `Reviews by category`, `Products by category`. 
2. Ensure controls display correctly.
3. Test searching for items in the list and ensure results are valid.
4. Test making selections, saving the post, and editing again to ensure selections are preserved.


## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Updated product category controls to use typescript
